sources/packages/backend/apps/db-migrations/src/sql/Reports/Create-offering-details-report.sql

```diff
@@ -0,0 +1,77 @@
+      education_programs.program_name AS "Program",
+      education_programs.sabc_code AS "SABC Program Code",
+      offering."Offering Name",
+      offering."Year of Study",
+      offering."Offering Intensity",
+      offering."Course Load",
+      offering."Delivery Type",
+      offering."WIL Component",
+      offering."WIL Component Type",
+      offering."Start Date",
+      offering."End Date",
+      offering."Has Study Breaks",
+      offering."Actual Tuition",
+      offering."Program Related Costs",
+      offering."Mandatory Fees",
+      offering."Exceptional Expenses",
+      offering."Public Offering",
+      offering."Status",
+      offering."Funded Weeks",
+      offering."Total Applications"
+    FROM
+      (
+      SELECT
+        education_programs_offerings.id AS "ID",
+        education_programs_offerings.offering_name AS "Offering Name",
+        education_programs_offerings.year_of_study AS "Year of Study",
+        education_programs_offerings."offering_intensity" AS "Offering Intensity",
+        education_programs_offerings.course_load AS "Course Load",
+        education_programs_offerings.offering_delivered AS "Delivery Type",
+        education_programs_offerings.has_offering_wil_component AS "WIL Component",
+        education_programs_offerings.offering_wil_type AS "WIL Component Type",
+        education_programs_offerings.study_start_date AS "Start Date",
+        education_programs_offerings.study_end_date AS "End Date",
+        NOT education_programs_offerings.lacks_study_breaks AS "Has Study Breaks",
+        education_programs_offerings.actual_tuition_costs AS "Actual Tuition",
+        education_programs_offerings.program_related_costs AS "Program Related Costs",
+        education_programs_offerings.mandatory_fees AS "Mandatory Fees",
+        education_programs_offerings.exceptional_expenses AS "Exceptional Expenses",
+        education_programs_offerings.offering_type AS "Public Offering",
+        education_programs_offerings."offering_status" AS "Status",
+        education_programs_offerings.study_breaks ->> ''totalFundedWeeks'' AS "Funded Weeks",
+        COUNT(applications.id) AS "Total Applications",
+        education_programs_offerings.location_id AS "Location ID",
+        education_programs_offerings.program_id AS "Program ID"
+      FROM
+        sims.education_programs_offerings education_programs_offerings
+      LEFT JOIN sims.student_assessments student_assessments ON
+        education_programs_offerings.id = student_assessments.offering_id
+      LEFT JOIN sims.applications applications ON
+        applications.id = student_assessments.application_id
+      LEFT JOIN sims.program_years program_years ON
+        applications.program_year_id = program_years.id
+      WHERE
+        program_years.id = :programYear
+        AND education_programs_offerings.offering_intensity = ANY(:offeringIntensity)
+      GROUP BY
+        education_programs_offerings.id) offering
```

Comment:
I did not find anything that would support that the `select` can support columns that are not part of the `group by` without possible unwanted effects. If you have something to support it, please share.
My recommendation would be having subquery to get only the total of offerings as filtered as possible.
```sql
(
	SELECT
		education_programs_offerings.id AS "offering_id",
		COUNT(applications.id) AS "total_offerings",
	FROM
		sims.education_programs_offerings education_programs_offerings
		inner join sims.program_years program_year on program_years.id = :programId
	WHERE
		education_programs_offerings.start_date between program_year.start_date and program_year.end_date
		AND education_programs_offerings.offering_intensity = ANY(:offeringIntensity)
		-- restrict also by the institution id.
	GROUP BY
		education_programs_offerings.id) total_offering
```

Reply:
I understand that you are talking about the most common group by aggregation error. Here, it is not the same case because I am grouping by the `ID` column. Database here knows that when grouping by the `ID` column, the select will only select one record in the group for all the columns in the `education_program_offering` table. If I would have tried grouping by the `offering name`,  then the aggregate group by error would have been thrown as shown below:

<img width="1091" alt="image" src="https://github.com/bcgov/SIMS/assets/7859295/e6f6d323-8603-477c-87b8-8d6458ea68f2">

Reply:
@andrewsignori-aot  Is below the expected subquery as per our discussion on the call:

<img width="1043" alt="image" src="https://github.com/bcgov/SIMS/assets/7859295/d4e6cc18-413d-46ed-b094-f71d77b9b58f">